builder/dockerfile/dispatchers.go

```diff
@@ -246,27 +246,19 @@ func (d *dispatchRequest) getImageOrStage(name string, platform *specs.Platform)
 		platform = d.builder.platform
 	}
 
-	// Windows cannot support a container with no base image unless it is LCOW.
+	// Windows cannot support a container with no base image.
 	if name == api.NoBaseImageSpecifier {
-		p := platforms.DefaultSpec()
-		if platform != nil {
-			p = *platform
+		// Windows supports scratch. What is not supported is running containers from it.
+		if runtime.GOOS == "windows" {
```

Comment:
Technically, shouldn't this be `platform.OS`?
(ie, this check moved down and put after we have `imageImage.OS` calculated so we can compare against that instead?)

It probably doesn't make much difference for how this code is actually used _today_, but it seems more "correct" (and thus safer for future potential use).

Reply:
Hm.. possibly we could use `platform` here. There's a lot of complex logic involved due to the combination of `platform` being passed as argument (optional), and the fallback to `d.builder.platform` (which also seems to be optional); https://github.com/moby/moby/blob/085c6a98d54720e70b28354ccec6da9b1b9e7fcf/builder/dockerfile/builder.go#L154-L162

So I tried not to diverge too much from the existing logic (for now), but agreed that some more of this needs to be looked at closely (as it's confusing where the options come from)